product-catalog/app/productcatalog/exceptions/ProductNotFoundException.java

```diff
@@ -0,0 +1,3 @@
+package productcatalog.exceptions;
+
+public class ProductNotFoundException extends Exception {}
```

Comment:
For a really good exception it would be cool to propagate some context. Maybe you want to get the product for a certain sku or a certain slug.